### PR TITLE
Font resources for Android

### PR DESCRIPTION
**Pull request by angpysha**
### Description of Change ###

Add the possibility to use Android font resources.
To use font resources you need to include font at app resources with Android resource name. Example
`
 <OnPlatform
            x:TypeArguments="x:String" x:Key="Exo2">
            <On Platform="Android" Value="exo2"></On>
        </OnPlatform>
`

### API Changes ###

Changed:
None
 

### Platforms Affected ### 
- Android

### Behavioral/Visual Changes ###
Add ability to use font resources for Xamarin Forms for Android instead assets

None

### Before/After Screenshots ### 
*Before*

*After*

### Testing Procedure ###
None

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by jfversluis** [comment]
Just curious, why would you want to use a font from the resources over a font from the assets? Is there an actual difference?

I also have [this PR](https://github.com/xamarin/Xamarin.Forms/pull/6013) in mind, where we want to overhaul the whole font loading and embedding story. Just thinking out loud here!

I appreciate your work, just trying to figure out if it's something we should add right now because of all the changes that are coming in this area and it might change again in the near future which would be a waste of your efforts

**Comment by angpysha** [comment]
The main advantage is the possibility to use extended font-family
```
<font-family xmlns:android="http://schemas.android.com/apk/res/android">
    <font
        android:fontStyle="normal"
        android:fontWeight="400"
        android:font="@font/lobster_regular" />
    <font
        android:fontStyle="italic"
        android:fontWeight="400"
        android:font="@font/lobster_italic" />
</font-family>
```
Also you can use remote fonts 
```
<?xml version="1.0" encoding="utf-8"?>
<font-family xmlns:android="http://schemas.android.com/apk/res/android"
    android:fontProviderAuthority="com.example.fontprovider.authority"
    android:fontProviderPackage="com.example.fontprovider"
    android:fontProviderQuery="Lobster"
    android:fontProviderCerts="@array/certs">
</font-family>
```

For more info, you can visit this link https://developer.android.com/guide/topics/resources/font-resource. I can also try to integrate my code to PR 6013

**Comment by samhouts** [comment]
@angpysha Are you able to address the feedback above? Thanks!

**Comment by angpysha** [comment]
@jfversluis thank you for testing. Sorry, I have passed exams at university. Now I can try to fix it.

**Comment by jfversluis** [comment]
No worries! Glad to hear you passed the exams, congratulations!

We'll be looking forward to your changes

**Comment by angpysha** [comment]
@jfversluis I have fixed it. To provide backward compatibility, I think the string of font must be like that "montserrat_medium#Montserrat-Medium". Such as for assets, but without the file extension.
Now it looks like this:


I used 2 missing fonts labels for these fonts:
"foo#bar"
"FooBar"

Maybe we need to change font string to another?

**Comment by jfversluis** [comment]
Build is failing with

> ##[error]C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1703,2): Error APT0000: invalid resource directory name: d:\agent\1\s\Xamarin.Forms.ControlGallery.Android\obj\Debug\90\res font "res font".

Does that mean anything to you?

And what do you mean by 

> Maybe we need to change font string to another?

**Comment by angpysha** [comment]
Strange error. I try to reproduce and fix it because on my machine is all right. 

> Maybe we need to change font string to another?

I ment that will be font string like this "montserrat_medium#Montserrat-Medium" good or maybe change it to another pattern?

**Comment by angpysha** [comment]
I tried to build on my CI and build successful.
https://dev.azure.com/angpysha/Xamarin%20forms%20my/_build/results?buildId=64

**Comment by jfversluis** [comment]
Hey @angpysha sorry for the long wait. I see there are some conflicts now. Would you be able to rebase so we can see to get this in sometime soon?

If you don't have the time, please let us know so we can take it from here.

Thanks!

**Comment by angpysha** [comment]
Hi @jfversluis. Sorry for long response. Yes, I can rebase or merge it. I will do it in this week

**Comment by jfversluis** [comment]
Great! Ping me when you do so I can have a look at it straight away when you do

**Comment by angpysha** [comment]
@jfversluis Merged

**Comment by jfversluis** [comment]
Thanks @angpysha! I see there is a couple of tests failing which don't seem related. I think we're working on those already so I think we have to hold on a little bit longer...

**Comment by jfversluis** [comment]
@angpysha again, so sorry. You just did the changes when the tests were in a bad state. Can I ask you to rebase one more time? If you can't do it, let me know and I will have a look myself. I can understand if you get tired of it at some point ;)

**Comment by angpysha** [comment]
@jfversluis Done

File: Stubs/Xamarin.Forms.Platform.Android/Xamarin.Forms.Platform.Android (Forwarders).csproj [hunk 1]
```diff
@@ -14,9 +14,15 @@
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <OutputPath>bin\$(Configuration)\$(TargetFramework)</OutputPath>
   </PropertyGroup>
+  <ItemGroup>
```
Comment (jfversluis) [inline Stubs/Xamarin.Forms.Platform.Android/Xamarin.Forms.Platform.Android (Forwarders).csproj:17]: Any idea why these entries were added? Also for other (unrelated) projects

**Comment by jfversluis** [comment]
I don't want to touch this too much for Forms anymore and I think the loading of fonts will change dramatically for .NET MAUI. I will close this one for now.

Thank you so much for your time and effort on this though!

**Pull request closed without merging**